### PR TITLE
Revert to preferring IP addresses for Node listeners by default

### PR DESCRIPTION
**Pull request by nightkr**
# Description

Supersedes #243. Implementation side of https://github.com/stackabletech/operator-rs/pull/903.

- [ ] Update https://github.com/stackabletech/kafka-operator/blob/87150b66d76b534b2859b9a58eb4f075f9ec7b31/docs/modules/kafka/examples/getting_started/kafka.yaml after the merge

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [x] CRD changes approved
- [x] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [x] Helm chart can be installed and deployed operator works
- [x] Integration tests passed (for non trivial changes)
- [x] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [x] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [x] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -34,3 +34,4 @@ tracing = "0.1.40"
 [patch."https://github.com/stackabletech/operator-rs.git"]
 # stackable-operator = { path = "../operator-rs/crates/stackable-operator" }
 # stackable-operator = { git = "https://github.com/stackabletech//operator-rs.git", branch = "main" }
+stackable-operator = { git = "https://github.com/stackabletech//operator-rs.git", branch = "feature/smart-preferred-address-type" }
```
Comment (sbernauer) [inline Cargo.toml:37]: reminder to switch once released

**Review by sbernauer — COMMENTED** [review]
Code LGTM. Can you please also update the docs?

**Comment by nightkr** [comment]
e1ff5b1bf3e97c60ffbd960d833a05ad69e6137d

**Review by sbernauer — DISMISSED** [review]
Thanks!

**Comment by adwk67** [comment]
Fixed Clusterrole:

```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
...
rules:
...
  resources:
  - persistentvolumes
  verbs:
  - get
  - list
  - watch
  - patch
  - create
  - delete
- apiGroups:
  - ""
  resources:
  - nodes
  - persistentvolumes
  - persistentvolumes/finalizers
  - persistentvolumeclaims
  - endpoints
  verbs:
  - create
  - update
  - patch
  - get
  - list
  - watch
...
- apiGroups:
  - listeners.stackable.tech
  resources:
  - listeners
  - listeners/finalizers
  - listeners/status
  - podlisteners
  verbs:
  - patch
  - create
  - delete
  - update
...
```

**Review by adwk67 — APPROVED** [review]
approved! will update the clusterrole for OS-compatability separately

**Pull request merged**